read-fonts/src/font_data.rs

```diff
@@ -172,11 +183,13 @@ impl<'a> FontData<'a> {
 
 impl<'a> Cursor<'a> {
     pub(crate) fn advance<T: Scalar>(&mut self) {
-        self.pos += T::RAW_BYTE_LEN
+        let advance_bytes = Saturating(T::RAW_BYTE_LEN.try_into().unwrap_or(u32::MAX));
+        self.pos += advance_bytes;
     }
 
     pub(crate) fn advance_by(&mut self, n_bytes: usize) {
-        self.pos += n_bytes;
+        let advance_bytes = Saturating(n_bytes.try_into().unwrap_or(u32::MAX));
+        self.pos += advance_bytes;
```

Comment:
hmm, okay... the original motivation for this design was that I wanted to reduce the number of bounds checks and branching during the initial parsing phase, so if this is actually generating code that includes a bounds check that doesn't feel great.

- That said, we don't have actual data to inform this decision, which would be nice.
- in the `advance` case, given that `RAW_BYTE_LEN` is `const` and the fn is generic, I would expect the compiler to elide the bounds check.
- we could probably make `RAW_BYTE_LEN` a u32 as well, if that is helpful?
- overall my main concern with this patch is that it makes it harder to understand the code that `rustc`/`llvm` are going to end up generating for us here..